### ISSUE TITLE
Program exiting with code 3489660927

### ISSUE THREAD

**Issue by dospunk** [gyscos/Cursive#159]
My program is crashing and exiting with code `3489660927`. I can't find anything about that code online and I'm fairly certain the crash is related to Cursive so does anyone know what it means?

**Comment by gyscos** [comment]
Thanks for the report!

I'm sorry you're having crashes. Sometimes the ncurses wrapper isn't quite a match with the system library and results in a segfaults, with weird error codes. Do you have a way to reproduce the error? If you can give me some code, I'll fire up a debugger it and see what's going on.

**Comment by dospunk** [comment]
currently I'm using the pancurses backend. I'll try to reproduce, I've been working on the program some more and haven't gotten the error again but I'll see if I can recreate it

**Comment by dospunk** [comment]
Oh yeah probably should have posted my specs

- OS: Windows 10 Home
- rustc version 1.22.1
- Using the most recent git version of Cursive

**Comment by gyscos** [comment]
If you run the IRC communication in a separate thread, and it doesn't affect stdin/stdout, it shouldn't interact too much with Cursive. 
Even if you run it in the same thread (stepping manually the cursive event loop), I don't think there would be any serious issue there.
And even if the IRC code was messing with the input/output, I think the worse case scenario would be a messed-up terminal state, but not a crash. Especially if it's not a clean rust panic. Such crash very likely happen in unsafe or native code, like pdcurses.

In case it is a rust panic, you can try running `export RUST_BACKTRACE=1` before starting the application, and redirect stderr to a file: `cargo run 2> error_output`, so you can check the file content for any error/backtrace if a crash happens again.

**Comment by kailando** [comment]
I'm having the same exit code in this program:

EDIT: I'm using Thonny IDE on Local Python 3. Also, change ".zip" to ".py".
[t.zip](https://github.com/gyscos/cursive/files/11131130/t.zip)

**Comment by gyscos** [comment]
@kailando Your issue appears unrelated. Your file includes this content:
```python
from turtle import *
import random
import pyautogui
mp=list(pyautogui.position())
print()
color("black")
begin_fill()
end_fill()
turtle.goto(mp)
done()
```
This repository is about a Rust TUI library, not the python turtle package.

**Comment by kailando** [comment]
Oh. I thought it was a [turtle] problem, not a Rust problem.

**Comment by kailando** [comment]
Yes, it was email. I know, why didn't I just pyt it here? Here's the awnser: I didn't know this was going to be sent to GitHub.

**Comment by Cyber-chan920** [comment]
If the canvas window loses control and you ask your computer to close it, this code will also appear.
Just stop the code and then run it again :)

![Image](https://github.com/user-attachments/assets/7338757a-55ab-4b1c-bd2e-ff79b141a0d4)

This is the code I was using if you need it:
`from turtle import*
import random
Turtle()
c=['blue','yellow','orange']
for i in range (200):
    color(random.choice(c))
    forward(i)
    left(144)
    speed(70)
`
